quickwit/quickwit-indexing/src/source/gcp_pubsub_source.rs

```diff
@@ -440,11 +515,182 @@ mod gcp_pubsub_emulator_tests {
             "index_id": index_id,
             "source_id": source_id,
             "subscription": subscription,
+            "partition_id": partition,
             "num_bytes_processed": 54,
             "num_messages_processed": 6,
             "num_invalid_messages": 0,
             "num_consecutive_empty_batches": 4,
         });
         assert_eq!(exit_state, expected_exit_state);
     }
+
+    #[tokio::test]
+    async fn test_gcp_pubsub_source_ack() {
+        let universe = Universe::with_accelerated_time();
+        let topic = append_random_suffix("test-gcp-pubsub-source--ack--topic");
+        let subscription = append_random_suffix("test-gcp-pubsub-source--ack--subscription");
+        let publisher = create_topic_and_subscription(&topic, &subscription).await;
+
+        let source_config = get_source_config(&subscription);
+        let index_id = append_random_suffix("test-gcp-pubsub-source--acl--index");
+        let metastore = metastore_for_test();
+        let index_uid = IndexUid::new(&index_id);
+        let (doc_processor_mailbox, _doc_processor_inbox) = universe.create_test_mailbox();
+
+        let ctx = SourceExecutionContext::for_test(
+            metastore,
+            index_uid,
+            PathBuf::from("./queues"),
+            source_config,
+        );
+        let mut gcp_source: GcpPubSubSource = GcpPubSubSource::try_new(
+            ctx,
+            GcpPubSubSourceParams {
+                subscription: subscription.clone(),
+                enable_backfill_mode: true,
+                credentials_file: None,
+                project_id: Some(GCP_TEST_PROJECT.to_string()),
+                max_messages_per_pull: None,
+            },
+        )
+        .await
+        .unwrap();
+
+        // we make a void actor. We will call the real source fn manually
+        let source_actor = SourceActor {
+            source: Box::new(VoidSource {}),
+            doc_processor_mailbox: doc_processor_mailbox.clone(),
+        };
+        let (source_mailbox, _source_handle) = universe.spawn_builder().spawn(source_actor);
+        let (observable_state_tx, _observable_state_rx) = watch::channel(json!({}));
+
+        let source_ctx = SourceContext::for_test(&universe, source_mailbox, observable_state_tx);
```

Comment:
We don't need to spawn a dummy source to get a source context for test. There's already `ActorContext::for_test`.